src/models/pieces/pawn.mjs

```diff
@@ -83,19 +93,47 @@ class Pawn extends Piece {
       available.push(Move.fromSquare(secondSquare, this));
     }
 
+    return available;
+  }
 
+  setPromote(move) {
+    if (move.row === 0 || move.row === 7) {
+      this.isPromoted = true;
+    }
+  }
 
-    return available;
-;
+  promote(move, type) {
```

Comment:
we'll probably want to move this out of here for the sake of avoiding circular dependencies.

Reply:
We'll probably want to create an `index.js` for the `pieces` package and export a utility `promote` function from there. We'll also want to export the pieces as well. This would be my TLDR suggestion. Further reading on [re-exporting/aggregating](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export#re-exporting_aggregating)

Reply:
Also, a nice readable function imo